### PR TITLE
HOFF-594 Implement internal server error page (500 status code)

### PR DESCRIPTION
**Pull request by TemitopeAyokuHO**
## What

[HOFF-594](https://collaboration.homeoffice.gov.uk/jira/browse/HOFF-594)- HTTP Error 500 (Internal Server Error) Error Page

## Why

To display an error page to users, which follows the Government Design Standard (GDS)

## How 

 - create a 500 error page in the view folder named 500.html, this page content the html element tag of content display
 - add an if block statement to capture the error code (UNKNOWN or INTERNAL_SERVER_ERROR) in the Middleware/errors.js file 
 - add an object of content to display in the Translations.en/errors.json file.

## Testing
Tested locally 
Screenshot 2024-02-25 at 22 27 45

Screenshot 2024-02-26 at 11 12 05

### CODE REVIEW

File: frontend/template-partials/translations/src/en/errors.json [hunk 1]
```diff
@@ -11,6 +11,14 @@
     "title": "Page not found",
     "description": "This page does not exist"
   },
+  "500": {
+    "title": "Internal Server Error",
+    "description": "Server is down",
+    "header": "Sorry, there is a problem with the service",
+    "paragraph1": "Try again later.",
+    "paragraph2": "We saved your answers. They will be available for 30 days.",
```
Comment (sulthan-ahmed) [inline frontend/template-partials/translations/src/en/errors.json:19]: This isn't true for most of our services

**Comment by sulthan-ahmed** [comment]
This content is too specific not all services have claim.  Who wrote the content for this ticket?

**Comment by TemitopeAyokuHO** [comment]
> This content is too specific not all services have claim. Who wrote the content for this ticket?
Actually, there was no content specified on the ticket to be displayed on the error page, and i can't leave that spot empty. So pending the time we can have a generic content on that section

File: frontend/template-partials/translations/src/en/errors.json (hunk 1 above)
Comment (TemitopeAyokuHO) [reply frontend/template-partials/translations/src/en/errors.json:19]: ATM there's no content specified for the ticket, i have just used the default GDS template and tweak some section. Please advice what generic content can we use

**Comment by sulthan-ahmed** [comment]
updated the ticket. Here is the suggested change unless a content designer has time to provide the content

Sorry, there is a problem with the service
Try again later.

File: components/address-lookup/index.js [hunk 2]
```diff
@@ -230,7 +230,7 @@ module.exports = config => {
                 }, req, res)
               };
             // cannot connect to validation service, skip api validation
-            } else if (err.status === 403 || err.status === 404) {
```
Comment (sulthan-ahmed) [inline components/address-lookup/index.js:233]: I don't we use this anymore

File: middleware/errors.js [hunk 3]
```diff
@@ -54,10 +50,22 @@ const getContent = (err, translate) => {
...
+    err.paragraph1 = (translate && translate('errors.500.paragraph1'));
+    err.paragraph2 = (translate && translate('errors.500.paragraph2'));
+    err.paragraph3 = (translate && translate('errors.500.paragraph3'));
+    content.title = (translate && translate('errors.500.title'));
+    content.message = (translate && translate('errors.500.description'));
+    content.header = (translate && translate('errors.500.header'));
+    content.paragraph1 = (translate && translate('errors.500.paragraph1'));
+    content.paragraph2 = (translate && translate('errors.500.paragraph2'));
+    content.paragraph3 = (translate && translate('errors.500.paragraph3'));
+  }
```
Comment (sulthan-ahmed) [inline middleware/errors.js:68]: not your fault but why does this all have to be set manually for each translation? 
can you just use a markdown file?

Comment (TemitopeAyokuHO) [reply middleware/errors.js:64]: Done

**Pull request closed without merging**